classes/license_image.bbclass

```diff
@@ -0,0 +1,281 @@
+ROOTFS_LICENSE_DIR = "${IMAGE_ROOTFS}/usr/share/common-licenses"
+
+python write_package_manifest() {
+    # Get list of installed packages
+    license_image_dir = d.expand('${LICENSE_DIRECTORY}/${IMAGE_NAME}')
+    bb.utils.mkdirhier(license_image_dir)
+    from oe.rootfs import image_list_installed_packages
+    from oe.utils import format_pkg_list
+
+    pkgs = image_list_installed_packages(d)
+    output = format_pkg_list(pkgs)
+    open(os.path.join(license_image_dir, 'package.manifest'),
+        'w+').write(output)
+}
+
+python license_create_manifest() {
+    import oe.packagedata
+    from oe.rootfs import image_list_installed_packages
+
+    skip_pkgs = set((d.getVar('COPY_LIC_DIRS_SKIP_PKGS') or "").split())
```

Comment:
I'm not sure I understand which issue this solution is fixing.

Is the problem that feed `-lic` packages might install their license information to anywhere in the rootfs - so BB doesn't know what content to copy into the common licenses directory, and therefore the tar archive? If so, how does skipping the NI-proprietary content resolve the issue? The NI IPKs shouldn't be more- or less-compliant than OE packages wrt. placing licenses in the correct locations.

Reply:
Adding onto Alex's question, it almost seems like we're saying "If its an NI package, its OK to ship without a license" ... and I'm not exactly sure that's right.  Maybe we should be filing bugs to the appropriate package owners to add a license to their packages?  Unless I'm completely misunderstanding the COPY_LIC_DIRS_SKIP_PKGS purpose.